src/main/kotlin/org/jetbrains/research/testspark/display/generatedTestsTab/GeneratedTestsTabBuilder.kt

```diff
@@ -118,16 +119,16 @@ class GeneratedTestsTabBuilder(
 
         report.testCaseList.values.forEach {
             val testCase = it
-            val testCasePanel = JPanel()
-            testCasePanel.layout = BorderLayout()
+            val testCaseDisplayPanel = JPanel()
+            testCaseDisplayPanel.layout = BorderLayout()
 
             // Add a checkbox to select the test
-            val checkbox = JCheckBox()
-            checkbox.isSelected = true
-            checkbox.addItemListener {
-                generatedTestsTabData.testsSelected -= (1 - 2 * checkbox.isSelected.compareTo(false))
+            val testCaseSelectedCheckbox = JCheckBox()
+            testCaseSelectedCheckbox.isSelected = true
+            testCaseSelectedCheckbox.addItemListener {
+                generatedTestsTabData.testsSelected -= (1 - 2 * testCaseSelectedCheckbox.isSelected.compareTo(false))
```

Comment:
This is a very tricky formula. Could you comment on it in the code and move the expression into a named variable so it is easy to understand what it does?

Reply:
It seems to be equivalent to:
```kotlin
generatedTestsTabData.testsSelected += (2 * testCaseSelectedCheckbox.isSelected.compareTo(false) - 1)
```